src/Controls/src/Core/Layout.cs

```diff
@@ -420,6 +417,14 @@ void InternalChildrenOnCollectionChanged(object sender, NotifyCollectionChangedE
 		{
 			if (e.Action == NotifyCollectionChangedAction.Move)
 			{
+				LogicalChildrenInternalBackingStore.RemoveAt(e.OldStartingIndex);
```

Comment:
Pretty much just a copy of this code
https://github.com/dotnet/corefx/blob/836f0480afbfc8fe7555530c6d98f70873db8ae4/src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs#L187